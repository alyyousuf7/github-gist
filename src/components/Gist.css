.gist .list-filetype {
    display: flex;
    flex-direction: row;
}

.gist .list-item-filetype {
    display: inline-flex;
    margin: 3px;
}

.gist .list-fork {
    display: flex;
    flex-direction: column;
}

.gist .list-item-fork {
    display: inline-flex;
}
