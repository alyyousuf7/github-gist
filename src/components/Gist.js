import PropTypes from 'prop-types';

import Badge from './Badge';
import UserBadge from './UserBadge';

import './Gist.css';

function Gist({ url, description, filetypes, forks }) {
  return <div className="gist">
    <a
      href={url}
      target="_blank"
      rel="noopener noreferrer"
      className="gist-description"
    >
      {description || '[no description]'}
    </a>
    <ul className="list-filetype">
      {filetypes.map((filetype, i) => <li key={i} className="list-item-filetype">
        <Badge>{filetype}</Badge>
      </li>)}
    </ul>
    <ul className="list-fork">
      {forks && forks.map((fork, i) => <li key={i} className="list-item-fork">
        <a
          href={fork.url}
          target="_blank"
          rel="noopener noreferrer"
          className="gist-description"
        >
          <UserBadge username={fork.owner.login} avatar={fork.owner.avatar} />
        </a>
      </li>)}
    </ul>
  </div>
}

Gist.propTypes = {
  url: PropTypes.string.isRequired,
  description: PropTypes.string,
  filetypes: PropTypes.arrayOf(PropTypes.string).isRequired,
  forks: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Gist;
