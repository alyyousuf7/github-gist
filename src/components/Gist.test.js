
import { render } from '@testing-library/react';

import Gist from './Gist';

describe('Gist', () => {
  it('should render correctly', async () => {
    const { container } = render(<Gist
      description='some description'
      url='some-url'
      filetypes={['type 1', 'type 2']}
      forks={[]}
    />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
