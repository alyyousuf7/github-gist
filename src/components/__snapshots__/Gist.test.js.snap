// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gist should render correctly 1`] = `
<div
  class="gist"
>
  <a
    class="gist-description"
    href="some-url"
    rel="noopener noreferrer"
    target="_blank"
  >
    some description
  </a>
  <ul
    class="list-filetype"
  >
    <li
      class="list-item-filetype"
    >
      <div
        class="badge"
      >
        type 1
      </div>
    </li>
    <li
      class="list-item-filetype"
    >
      <div
        class="badge"
      >
        type 2
      </div>
    </li>
  </ul>
  <ul
    class="list-fork"
  />
</div>
`;
